src/lib/pipeline.ts

```diff
@@ -175,6 +177,14 @@ function simplifyMongoPipeline(mongoSteps: Array<MongoStep>): Array<MongoStep> {
 
   for (const step of mongoSteps.slice(1)) {
     if (step.$project !== undefined && lastStep.$project !== undefined) {
+      for (const key in step.$project) {
+        // We do not want to merge two $project with common keys
+        if (key in lastStep.$project) {
```

Comment:
```suggestion
        if (lastStep.$project.hasOwnProperty(key)) {
```

In our cases, the `in` operator should perfectly do the job but there might be some subtle differences if at some point we get an object extending another one's prototoype. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty